src/include/platform/internal/GenericPlatformManagerImpl_Zephyr.ipp

```diff
@@ -109,14 +114,36 @@ void GenericPlatformManagerImpl_Zephyr<ImplClass>::_UnlockChipStack(void)
 template <class ImplClass>
 CHIP_ERROR GenericPlatformManagerImpl_Zephyr<ImplClass>::_StartChipTimer(System::Clock::Timeout delay)
 {
-    // Let Systemlayer.PrepareEvents() handle timers.
+#if !CHIP_SYSTEM_CONFIG_USE_SOCKETS
+    // If the platform timer is being updated by a thread other than the event loop thread,
+    // trigger the event loop thread to recalculate its wait time by posting a no-op event
+    // to the event queue.
+    if (k_current_get() != &mChipThread)
+    {
+        ChipDeviceEvent noop{ .Type = DeviceEventType::kNoOp };
+        ReturnErrorOnFailure(Impl()->PostEvent(&noop));
+    }
+#endif // !CHIP_SYSTEM_CONFIG_USE_SOCKETS
+
     return CHIP_NO_ERROR;
 }
 
 template <class ImplClass>
 CHIP_ERROR GenericPlatformManagerImpl_Zephyr<ImplClass>::_StopEventLoopTask(void)
 {
     mShouldRunEventLoop = false;
+
+#if !CHIP_SYSTEM_CONFIG_USE_SOCKETS
+    // If the platform timer is being updated by a thread other than the event loop thread,
```

Comment:
This comment seems copy-pasted but is not relevant for this method (stopping the CHIP thread).